### PR TITLE
feat(secret): add ci and doc for some auth plugins

### PR DESCRIPTION
**Pull request by soulbird**
### Description

As part of #8319:

After #8421, all authentication plugins support using APISIX Secret, but only the key-auth plugin adds test case coverage.
Here we have added test cases for basic-auth, hmac-auth, and wolf-rbac plugins using APISIX Secret resources to improve related functions. In addition, the documentation of related plugins is also supplemented.

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: t/plugin/basic-auth.t [hunk 1]
```diff
@@ -452,3 +452,99 @@ GET /echo
...
+            local t = require("lib.test_admin").test
+            -- put secret vault config
+            local code, body = t('/apisix/admin/secrets/vault/test1',
+                ngx.HTTP_PUT,
+                [[{
+                    "uri": "http://127.0.0.1:8200",
+                    "prefix" : "kv/apisix",
+                    "token" : "root"
+                }]],
+                [[{
```
Comment (spacewander) [inline t/plugin/basic-auth.t:473]: We don't need to check the response, as it is checked in the t/admin tests.

File: t/plugin/hmac-auth3.t [hunk 2]
```diff
@@ -755,3 +755,146 @@ apisix:
...
+            local t = require("lib.test_admin").test
+            -- put secret vault config
+            local code, body = t('/apisix/admin/secrets/vault/test1',
+                ngx.HTTP_PUT,
+                [[{
+                    "uri": "http://127.0.0.1:8200",
+                    "prefix" : "kv/apisix",
+                    "token" : "root"
+                }]],
+                [[{
```
Comment (spacewander) [inline t/plugin/hmac-auth3.t:776]: Ditto

File: t/plugin/hmac-auth3.t [hunk 3]
```diff
@@ -755,3 +755,146 @@ apisix:
...
+                    }
+                }]]
+                )
+            if code >= 300 then
+                ngx.status = code
+                return ngx.say(body)
+            end
+
+            -- set route
+            code, body = t('/apisix/admin/routes/1',
```
Comment (spacewander) [inline t/plugin/hmac-auth3.t:810]: Could you move the new tests to a new file?

> When the test file is too large, for example > 800 lines, you should split it to a new file. Please take a look at t/plugin/limit-conn.t and t/plugin/limit-conn2.t.

https://github.com/apache/apisix/blob/master/CONTRIBUTING.md#check-code-style-and-test-case-style

**Review by spacewander — COMMENTED** [review]
Some auth plugins like ldap-auth and jwt-auth don't exist in this change?

**Comment by soulbird** [comment]
> Some auth plugins like ldap-auth and jwt-auth don't exist in this change?

I missed it, I'll add it later

File: t/plugin/hmac-auth3.t (hunk 3 above)
Comment (soulbird) [reply t/plugin/hmac-auth3.t:810]: OK

File: docs/zh/latest/plugins/jwt-auth.md [hunk 4]
```diff
@@ -42,9 +42,9 @@ Consumer 端：
...
 | ------------- | ------- | ----- | ------- | --------------------------- | ------------------------------------------------------------------------------------------------------------ |
 | key           | string  | 是    |         |                             | Consumer 的 `access_key` 必须是唯一的。如果不同 Consumer 使用了相同的 `access_key` ，将会出现请求匹配异常。 |
-| secret        | string  | 否    |         |                             | 加密秘钥。如果未指定，后台将会自动生成。                                                                  |
-| public_key    | string  | 否    |         |                             | RSA 或 ECDSA 公钥， `algorithm` 属性选择 `RS256` 或 `ES256` 算法时必选。                                                            |
-| private_key   | string  | 否    |         |                             | RSA 或 ECDSA 私钥， `algorithm` 属性选择 `RS256` 或 `ES256` 算法时必选。                                                            |
+| secret        | string  | 否    |         |                             | 加密秘钥。如果未指定，后台将会自动生。成。该字段支持使用 [APISIX Secret](../terminology/secret.md) 资源，将值保存在 Secret Manager 中。   |
```
Comment (spacewander) [inline docs/zh/latest/plugins/jwt-auth.md:45]: Typo

File: t/plugin/hmac-auth4.t [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    }
+});
+
+run_tests;
+
+__DATA__
+
+=== TEST 1: set hmac-auth conf: secret_key uses secret ref
+--- request
+GET /t
```
Comment (spacewander) [inline t/plugin/hmac-auth4.t:43]: We already specified it at the top?

File: t/plugin/hmac-auth4.t [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+            nil,
+            headers
+        )
+
+        ngx.status = code
+        ngx.say(body)
+    }
+}
+--- request
+GET /t
```
Comment (spacewander) [inline t/plugin/hmac-auth4.t:170]: Ditto

File: t/plugin/jwt-auth3.t [hunk 7]
```diff
@@ -442,3 +457,201 @@ IeQmD/K+DClZMqSrliUzUqJnCPCzy6kCIAekDsRh/UF4ONjAJkKuLedDUfL3rNFb
...
+            -- enable jwt auth plugin using admin api
+            local code, body = t('/apisix/admin/routes/1',
+                ngx.HTTP_PUT,
+                [[{
+                    "plugins": {
+                        "jwt-auth": {}
+                    },
+                    "upstream": {
+                        "nodes": {
+                            "127.0.0.1:8777": 1
```
Comment (spacewander) [inline t/plugin/jwt-auth3.t:576]: Why don't we use the regular 1980 and /hello?

**Review by spacewander — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

**Pull request merged**